### PR TITLE
Documentation fixes and exclude private members from public API documentation

### PR DESCRIPTION
**Pull request by richnew10**
I've been applying some tooling, and found a bunch of broken links, malformed syntax, etc. in our documentation comments.

This PR fixes essentially every one of them.

Additionally, this PR adds the `--excludePrivate` flag to our TypeDoc configuration, which prevents private members and methods from being included in generated documentation. This is a significant improvement in clarity.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by richnew10** [comment]
One problem here is that TypeDoc doesn't actually support the TSDoc standard. For example TSDoc says that params should have a hyphen:

```typescript
/**
 * Frob the bar.
 * @param foo - the foo to use.
 */
function frob(foo: number): void {
}
```

while TypeDoc expects there to be no hyphen, and actually renders it into the doc output.

I'm going to split up this PR into fixes that will work with TypeDoc so we can land them now, and keep the rest for later.

**Comment by richnew10** [comment]
Content Share test passed (as you would expect — there are no code changes in this PR) but ended with a cancel.

File: src/signalingprotocol/SignalingProtocol.d.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class SdkSignalFrame implements ISdkSignalFrame {
 
     /**
      * Constructs a new SdkSignalFrame.
-     * @param [properties] Properties to set
+     * @param properties Properties to set
```
Comment (XHatan) [inline src/signalingprotocol/SignalingProtocol.d.ts:69]: is this change manual ? SignalingProtocol.d.ts is auto-generated through script/generate-signaling-protocol, maybe we should update the script ?

Comment (richnew10) [reply src/signalingprotocol/SignalingProtocol.d.ts:69]: It is a manual change.

The protobufjs version used in that script is outdated, as is its dependency on jsdoc — and neither of them produce the correct output.

generate-signaling-protocol isn't run automatically.

I will make a note to write a postprocessing script as a follow-up to make it generate correct output.

File: src/backoff/BackoffFactory.ts [hunk 2]
```diff
@@ -6,13 +6,11 @@ import Backoff from './Backoff';
 export default interface BackoffFactory {
   /**
    * Backoff factory method
-   * @returns {Backoff}
```
Comment (devalevenkatesh) [inline src/backoff/BackoffFactory.ts:9]: This is returning `Backoff` but, we are removing this?

File: src/datamessage/DataMessage.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export default class DataMessage {
   /**
    * true if server throttled or rejected message
    * false if server has posted the message to its recipients
-   * Throttling soft limit: Rate:100, Burst:200
+   * Throttling soft limit: Rate: 100, Burst:200
```
Comment (devalevenkatesh) [inline src/datamessage/DataMessage.ts:34]: Need space in front of 200 as well here, right?

Burest: 200.

File: src/signalingclient/SignalingClient.ts [hunk 4]
```diff
@@ -38,29 +38,29 @@ export default interface SignalingClient {
...
-   * @param {SignalingClientJoin} settings How to configure the Join frame.
+   * @param settings How to configure the Join frame.
    */
   join(settings: SignalingClientJoin): void;
 
   /**
    * Sends a subscribe frame with the given settings.
    *
-   * @param {SignalingClientSubscribe} settings How to configure the Subscribe frame.
+   * @param settings How to configure the Subscribe frame.
```
Comment (devalevenkatesh) [inline src/signalingclient/SignalingClient.ts:63]: So, does it still display the parameter type name or its fine as we already show in the method signature. Does it still highlight or linkify if users want to click and go to it?

We are removing in other classes as well so just trying to understand the end output here.

File: typedoc.json [hunk 5]
```diff
@@ -1,6 +1,7 @@
 {
   "out": "./docs",
   "readme": "README.md",
+  "excludePrivate": true,
   "exclude": [
     "src/signalingprotocol/SignalingProtocol.js",
     "src/signalingprotocol/SignalingProtocol.d.ts",
```
Comment (devalevenkatesh) [inline typedoc.json:7]: Since these two files are excluded should we make changes to them?

Comment (richnew10) [reply typedoc.json:7]: They are still processed by analysis tools like api-extractor, so I fixed them.

File: src/signalingclient/SignalingClient.ts (hunk 4 above)
Comment (richnew10) [reply src/signalingclient/SignalingClient.ts:63]: The syntax I'm fixing here is JSDoc. JavaScript doesn't have a way to type-annotate parameters, so it uses this `{syntax}` to indicate the type.

TypeScript has type annotations, so TypeDoc and other tools _already know_ that `settings` is a `SignalingClientSubscribe` instance, and can link directly to the correct type.

The output is exactly the same; TypeDoc is ignoring the `{SignalingClientSubscribe}` here.

Comment (richnew10) [reply src/signalingclient/SignalingClient.ts:63]: Here's what it looks like.

Screen Shot 2021-04-08 at 2 45 46 PM

Comment (devalevenkatesh) [reply src/signalingclient/SignalingClient.ts:63]: Cool, thanks for the explanation!

File: typedoc.json (hunk 5 above)
Comment (devalevenkatesh) [reply typedoc.json:7]: Got it, so no configuration on the api-extractor level in that case?

Comment (richnew10) [reply typedoc.json:7]: We'll get there when we get there :joy:

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**